synapse/rest/media/v1/_base.py

```diff
@@ -96,7 +113,15 @@ def add_file_headers(request, media_type, file_size, upload_name):
     def _quote(x):
         return urllib.parse.quote(x.encode("utf-8"))
 
-    request.setHeader(b"Content-Type", media_type.encode("UTF-8"))
+    # will only fire for unspecified charsets, i.e.,
+    # 'text/css' but not 'text/css; charset=UTF-16'
+    content_type = (
```

Comment:
the ternary feels awkward here.

Can we just do:

```python
if media_type.lower() in TEXT_CONTENT_TYPES:
    media_type += "; charset=UTF-8"

request.setHeader(b"Content-Type", media_type.encode("UTF-8"))

Reply:
Absolutely. I removed the ternary. However, I think we should keep the separate `content_type` variable because `media_type` is as an input variable. Generally speaking, you want to always have the original input variables for later use, and make new variables for derived traits like the `content_type` that will ultimately served to the user.